doc/src/guides/solving/solve-diophantine-equation.md

```diff
@@ -0,0 +1,207 @@
+# Solve a Diophantine Equation Algebraically
+
+Use SymPy to solve a Diophantine equation (find integer solutions to a
+polynomial equation) algebraically, returning a parameterized general solution
+if possible. For example, solving the [Pythagorean
+theorem](https://en.wikipedia.org/wiki/Pythagorean_theorem) $a^2 + b^2 = c^2$
+yields $(a=2pq, b=p^2-q^2, c=p^2-q^2)$.
+
+## Alternatives to Consider
+
+There are few alternatives for finding a parameterized general solution a
+Diophantine equation.
+- Numerical alternatives:
+    - [Sage's EllipticCurve
+  command](https://doc.sagemath.org/html/en/constructions/elliptic_curves.html)
+  may be able to find a set of relative numerical values for each variable
+    - You can test explicit integer values, for example using a nested for loop
+  of ranges of values. This is inefficient, but fine if you are only interested
+  in solutions that are relatively small.
+- {func}`~.solve` simply solves for one variable in terms of the others. For
+example, attempting to solve $a^2 + b^2 = c^2$ for $a$, $b$, and $c$ can only
+reveal that $a = \pm \sqrt{c^2-b^2}$:
+
+```py
+>>> from sympy import solve
+>>> from sympy import symbols
+>>> a, b, c = symbols("a, b, c", integer=True)
+>>> solve(a**2 + b**2 - c**2, [a, b, c], dict=True)
+[{a: -sqrt(-b**2 + c**2)}, {a: sqrt(-b**2 + c**2)}]
+```
+
+## Example of Solving a Diophantine Equation
+
+Here is an example of solving a Diophantine equation, specifically $a^2 + b^2 =
+c^2$ transformed to ***, using {func}`~.diophantine`:
+
+```py
+>>> from sympy.solvers.diophantine import diophantine
+>>> from sympy import symbols
+>>> a, b, c = symbols("a, b, c", integer=True)
+>>> my_syms = (a, b, c)
+>>> pythag = a**2 + b**2 - c**2
+>>> d = diophantine(pythag, syms=my_syms)
+>>> d
+{(2*p*q, p**2 - q**2, p**2 + q**2)}
+```
+
+Refer to the [Diophantine API reference](../../modules/solvers/diophantine.rst)
+for more examples of solving various types of Diophantine equations.
+
+## Guidance
+
+### Expression to Solve Must Equal Zero
```

Comment:
It looks like diophantine accepts Eq(), similar to solve. This makes it sound like it doesn't, but that is also an option if you don't want to subtract one side of the equation.